pkg/common/cns-lib/vsphere/utils.go

```diff
@@ -460,3 +461,39 @@ func IsvSphereVersion70U3orAbove(ctx context.Context, aboutInfo types.AboutInfo)
 	// For all other versions.
 	return false, nil
 }
+
+// IsVolumeCreationSuspended checks whether a given Datastore has cns.vmware.com/datastoreSuspended customValue
+func IsVolumeCreationSuspended(datastoreInfo *DatastoreInfo, cnsMgrSuspendCreateVolumeFSS bool) bool {
+
+	if cnsMgrSuspendCreateVolumeFSS == false {
+		return false
+	}
+
+	for _, customValField := range datastoreInfo.CustomValue {
+		customVal := customValField.(*types.CustomFieldStringValue)
+		if customVal.Value == cnsMgrDatastoreSuspended {
+			return true
+		}
+	}
+
+	return false
+}
+
+// FilterSuspendedDatastoreInfo filters out datastores which cns.vmware.com/datastoreSuspended customValue
+func FilterSuspendedDatastoreInfo(ctx context.Context, datastoreInfoList []*DatastoreInfo, cnsMgrSuspendCreateVolumeFSS bool) []*DatastoreInfo {
```

Comment:
Same. It shouldn't check for `cnsMgrSuspendCreateVolumeFSS` value to be true. It should be done in calling function.